var/spack/repos/builtin/packages/gdal/package.py

```diff
@@ -155,7 +155,8 @@ class Gdal(AutotoolsPackage):
     depends_on('python@2.0:', type=('build', 'link', 'run'), when='@3.2:+python')
     depends_on('python', type=('build', 'link', 'run'), when='+python')
     # swig/python/setup.py
-    depends_on('py-setuptools', type='build', when='+python')
+    depends_on('py-setuptools@:57', type='build', when='@:3.0.4+python')
+    depends_on('py-setuptools', type='build', when='@3.1.0:+python')
```

Comment:
```suggestion
    depends_on('py-setuptools@:57', type='build', when='@:3.0+python')
    depends_on('py-setuptools', type='build', when='@3.1:+python')
```
In case they someday add a 3.0.5 release.